File: test/ebay_image_url_test.dart
import 'package:blindbox_app/features/market/utils/ebay_image_url.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('upgradeEbayImageUrl upgrades browse thumbs', () {
    expect(
      upgradeEbayImageUrl('https://i.ebayimg.com/g/s-l225.jpg'),
      'https://i.ebayimg.com/g/s-l500.jpg',
    );
  });

  test('upgradeEbayImageUrl upgrades detail thumbs', () {
    expect(
      upgradeEbayImageUrl(
        'https://i.ebayimg.com/g/s-l500.jpg',
        size: EbayImageSize.detail,
      ),
      'https://i.ebayimg.com/g/s-l1600.jpg',
    );
  });

  test('ebayBrowseItemId wraps legacy ids', () {
    expect(ebayBrowseItemId('110589358256'), 'v1|110589358256|0');
    expect(ebayBrowseItemId('v1|110589358256|0'), 'v1|110589358256|0');
  });
}

File: functions/test/ebay-image-url.test.js
'use strict';

const { describe, it } = require('node:test');
const assert = require('node:assert/strict');
const {
  pickBestEbayImageUrl,
  upgradeEbayImageUrl,
} = require('../lib/providers/gateway/ebayImageUrl');

describe('upgradeEbayImageUrl', () => {
  it('upgrades browse thumbs to s-l500', () => {
    const url =
      'https://i.ebayimg.com/images/g/abc/s-l225.jpg';
    assert.equal(
      upgradeEbayImageUrl(url, 'browse'),
      'https://i.ebayimg.com/images/g/abc/s-l500.jpg',
    );
  });

  it('upgrades detail thumbs to s-l1600', () => {
    const url =
      'https://i.ebayimg.com/images/g/abc/s-l500.jpg';
    assert.equal(
      upgradeEbayImageUrl(url, 'detail'),
      'https://i.ebayimg.com/images/g/abc/s-l1600.jpg',
    );
  });
});

describe('pickBestEbayImageUrl', () => {
  it('prefers the largest s-l token', () => {
    const raw = {
      thumbnailImages: [
        { imageUrl: 'https://i.ebayimg.com/a/s-l225.jpg' },
        { imageUrl: 'https://i.ebayimg.com/b/s-l500.jpg' },
      ],
    };
    assert.equal(
      pickBestEbayImageUrl(raw),
      'https://i.ebayimg.com/b/s-l500.jpg',
    );
  });
});
